File: lecture006/arrSubsets.cpp
#include<iostream>
#include<vector>
using namespace std;

void print(vector<int> &arr , int si , int ei){
    for(int i=si ; i<=ei ; i++){
            cout<<arr[i]<<" ";
    }
    cout<<endl;
}
void Substs(vector<int> &arr){
    for(int i=0 ; i<arr.size() ; i++){
        for(int j=i ; j<arr.size() ; j++){
             print(arr , i , j);
            //cout<<arr[j];
        }
       // cout<<endl;
    }  
}
int main(int args , char**argv){
    int n;
    cout<<"Enter size of array : ";
    cin>>n;
    vector<int> arr(n,0);
    cout<<"enter array : ";
    for(int i=0 ; i<n ; i++){
        cin>>arr[i];
    }
    Substs(arr);
    return 0;
}

File: lecture006/arrSubsets.java
import java.util.Scanner;

public class arrSubsets {
    public static Scanner scn = new Scanner(System.in);

    public static void main(String[] args) {
        System.out.print("enter size of array : ");
        int n = scn.nextInt();
        int[] arr = new int[n];
        System.out.print("enter array ");
        for (int i = 0; i < arr.length; i++) {
            arr[i] = scn.nextInt();
        }
        Subsets(arr);
    }

    public static void Subsets(int[] arr) {
        for (int i = 0; i < arr.length; i++) {
            for (int j = i; j < arr.length; j++) {
                print(arr, i, j);
            }
        }

    }

    public static void print(int[] arr, int si, int ei) {
        for (int i = si; i < ei; i++) {
            System.out.print(arr[i]);
        }
        System.out.println();
    }
}